app/code/Magento/Version/Controller/Index/Index.php

```diff
@@ -22,12 +22,22 @@ class Index extends Action
     protected $productMetadata;
 
     /**
+     * @var State
+     */
+    protected $state;
+
+    /**
      * @param Context $context
      * @param ProductMetadataInterface $productMetadata
+     * @param State $state
      */
-    public function __construct(Context $context, ProductMetadataInterface $productMetadata)
-    {
+    public function __construct(
+        Context $context,
+        ProductMetadataInterface $productMetadata,
+        State $state
```

Comment:
This will need to be injected in with an optional dependency. See http://devdocs.magento.com/guides/v2.2/contributor-guide/backward-compatible-development/index.html#adding-a-constructor-parameter